crates/cli/src/tasks/csharp.rs

```diff
@@ -74,19 +74,33 @@ pub(crate) fn build_csharp(project_path: &Path, build_debug: bool) -> anyhow::Re
     } else {
         "AppBundle"
     };
-    let mut output_path = project_path.join(format!("bin/{config_name}/net8.0/wasi-wasm/{subdir}/StdbModule.wasm"));
-    if !output_path.exists() {
-        // check for the old .NET 7 path for projects that haven't migrated yet
-        output_path = project_path.join(format!("bin/{config_name}/net7.0/StdbModule.wasm"));
+
+    // check for the old .NET 7 path for projects that haven't migrated yet
+    let bad_output_paths = [
+        project_path.join(format!("bin/{config_name}/net7.0/StdbModule.wasm")),
+        // for some reason there is sometimes a tilde here?
+        project_path.join(format!("bin~/{config_name}/net7.0/StdbModule.wasm")),
+    ];
+    if bad_output_paths.iter().any(|p| p.exists()) {
+        anyhow::bail!(concat!(
+            "Looks like your project is using the deprecated .NET 7.0 WebAssembly bindings.\n",
+            "Please migrate your project to the new .NET 8.0 template and delete the folders: bin, bin~, obj, obj~"
+        ));
+    }
+    let possible_output_paths = [
+        project_path.join(format!("bin/{config_name}/net8.0/wasi-wasm/{subdir}/StdbModule.wasm")),
+        project_path.join(format!("bin~/{config_name}/net8.0/wasi-wasm/{subdir}/StdbModule.wasm")),
+    ];
+    if possible_output_paths.iter().all(|p| p.exists()) {
+        anyhow::bail!(concat!(
+            "For some reason, your project has both a `bin` and a `bin~` folder.\n",
+            "I don't know which to use, so please delete both and rerun this command so that we can see which is up-to-date."
+        ));
```

Comment:
should this be a first step? Like if both `bin` and `bin~` exist, then maybe _one_ of them is outdated, so the above "you're using deprecated bindings" messages might be erroneous.

Reply:
that said, I'm surprised that we newly need this behavior. I know we hardcode the C#/Unity SDK project itself to generate `bin~` so that the output isn't picked up by Unity (which ignores things with a `~` suffix) but I didn't think that leaked into users' projects..